src/vectorspace.js

```diff
@@ -0,0 +1,70 @@
+module.exports = {averageVectorMinDistance, averageVectorDistance};
+const {
+    assertArray,
+    assertArrayItemsInstanceOf,
+    assertArrayLengthEq,
+    assertArraysOfArrayNotEmpty,
+    assertArrayNotEmpty,
+    assertNumber
+} = require("./typeassert");
+const {lpDistance} = require("./math.js");
+
+/**
+ * @summary Measures distribution of vectors in vectorspace by summing minimum distances
+ * @param {number[][]} criteria 2d array with students and their criterias
+ * @param {number} p p value for lpdistance function
+ * @param {Function} weighted function that adds a weight
+ * @returns {number} the calculated score
+ */
+function averageVectorMinDistance(criteria, p, weighted){
+    assertArray(criteria);
+    assertArrayNotEmpty(criteria);
+    assertArrayItemsInstanceOf(criteria, Array);
+    assertArraysOfArrayNotEmpty(criteria);
+    assertArrayLengthEq(...criteria);
+    assertNumber(p);
+
+    let score = 0;
+    let totalDistances = 0;
+    for (let criteria1 = 0; criteria1 < criteria.length; criteria1++) {
+        let minDist = Infinity;
+        for (let criteria2 = 0; criteria2 < criteria.length; criteria2++) {
+            if (criteria1 !== criteria2){
+                minDist = weighted(Math.min(minDist, lpDistance(criteria[criteria1], criteria[criteria2], p)));
```

Comment:
Her kan javascripts indbyggede funktioner nok også benyttes

Reply:
criteria1 og criteria2 er ulæseligt, især når de benyttes som index i criteria arrayet

Reply:
Vi har omnavngivet criteria1 og criteria2 til i og j. Vi vælger ikke at bruge javascript